src/streaming-services/common/ScrobbleEvents.ts

```diff
@@ -22,30 +51,77 @@ export abstract class ScrobbleEvents {
 		}
 	}
 
-	getLocation(): string {
-		return window.location.href;
+	protected async checkForChanges(): Promise<void> {
+		const newUrl = this.getLocation();
+		if (this.url !== newUrl) {
+			await this.onUrlChange(this.url, newUrl);
+			this.url = newUrl;
+		}
+
+		const playback = await this.parser.parsePlayback();
+		if (playback) {
+			const newProgress = playback.progress;
+			if (this.progress !== newProgress) {
+				await this.updateProgress(newProgress);
```

Comment:
Because it now updates the progress before starting the scrobble, I'm pretty sure this fixes #73.

Reply:
Wait, actually it still wouldn't work because `TraktItem` wouldn't be set. I'll fix it by getting the progress from `Item` before starting the scrobble.

Reply:
Alright, now it should work properly.